### ISSUE TITLE
Doesn't work inside a Dialog component

### ISSUE THREAD

**Issue by jackiepapers** [sersavan/shadcn-multi-select-component#27]
Clicking first item has no effect, clicking second item closes dialog. Clicking close button also closes dialog.

**Comment by ryanschiang** [comment]
This component uses a Popover, so when wrapping it in a Dialog, you need to set `modalPopover={true}` as per [https://github.com/radix-ui/primitives/issues/2121#issuecomment-2227348602](https://github.com/radix-ui/primitives/issues/2121#issuecomment-2227348602)
```
<MultiSelect
    modalPopover={true}
    options={...}
    onValueChange={...}
/>
```

**Comment by wflkaaa** [comment]
> This component uses a Popover, so when wrapping it in a Dialog, you need to set `modalPopover={true}` as per [radix-ui/primitives#2121 (comment)](https://github.com/radix-ui/primitives/issues/2121#issuecomment-2227348602)
> 
> ```
> <MultiSelect
>     modalPopover={true}
>     options={...}
>     onValueChange={...}
> />
> ```

thanks, this solves the issue, but the search and scrollbar of multi-select still not working when inside dialog

**Comment by zaidrashid** [comment]
> > This component uses a Popover, so when wrapping it in a Dialog, you need to set `modalPopover={true}` as per [radix-ui/primitives#2121 (comment)](https://github.com/radix-ui/primitives/issues/2121#issuecomment-2227348602)
> > ```
> > <MultiSelect
> >     modalPopover={true}
> >     options={...}
> >     onValueChange={...}
> > />
> > ```
> 
> thanks, this solves the issue, but the search and scrollbar of multi-select still not working when inside dialog

Yes, scrolling and searching now does not work. Is there a work around for this?

**Comment by m5khan** [comment]
**To fix the search** I did `modal={false}` on `<Sheet />` or `<Dialog/>` and also  `modalPopover={false}` on MultiSelect Component
```
<Dialog
  modal={false}
>
   ...
    <MultiSelect
        modalPopover={false}
        options={...}
        onValueChange={...}
    />
  ...
</Dialog>
```

**Comment by paulotolentino** [comment]
Thanks @m5khan !! this approach worked just fine!

**Comment by vamnguyen** [comment]
### Thank @m5khan so much! I fixed it <3 **(2025/04/29)**:
**My package.json version:**
"@radix-ui/react-dialog": "^1.1.11",
"@radix-ui/react-popover": "^1.1.11",

**Comment by ws-t-jer** [comment]
I resolved the scroll with the following code:
```
  <PopoverContent
    ...
    onWheel={(e) => {
      e.stopPropagation();
    }}
    onTouchMove={(e) => {
      e.stopPropagation();
    }}
  >
```
https://github.com/radix-ui/primitives/issues/1159

**Comment by BaraJabali** [comment]
for the search issue use PopoverContentDialog instead of PopoverContent inside the component

**Comment by arrizkyhp** [comment]
try below approach, it's worked for me
https://github.com/shadcn-ui/ui/issues/1511#issuecomment-1784645453
